core/GlobalState.cc

```diff
@@ -1771,6 +1771,18 @@ void GlobalState::deleteMethodSymbol(MethodRef what) {
     this->methods[what.id()] = this->methods[0].deepCopy(*this);
 }
 
+void GlobalState::deleteFieldSymbol(FieldRef what) {
+    ENFORCE(what.data(*this)->flags.isField);
+    const auto &whatData = what.data(*this);
+    auto owner = whatData->owner;
+    auto &ownerMembers = owner.data(*this)->members();
+    auto fnd = ownerMembers.find(whatData->name);
+    ENFORCE(fnd != ownerMembers.end());
+    ENFORCE(fnd->second == what);
+    ownerMembers.erase(fnd);
+    this->fields[what.id()] = this->fields[0].deepCopy(*this);
+}
```

Comment:
slightly annoying that this logic is exactly the same as the method code, except for the types 🤔 no change expected just food for thought.

Reply:
I was `:sadpanda:` about this too, but I'm not sure there's a great way around it atm.  I guess we could have a helper like:

```c++
template <typename SymbolType, typename DataType>
void deleteSymbol(GlobalState &gs, SymbolType what, vector<DataType> &datavec) {
  ...
}

void GlobalState::deleteFieldSymbol(FieldRef what) {
  deleteSymbol(*this, what, this->fields);
}
```

I don't know if that's better or worse.